kratos/solving_strategies/schemes/residual_based_bossak_displacement_scheme.hpp

```diff
@@ -160,84 +145,68 @@ class ResidualBasedBossakDisplacementScheme: public Scheme<TSparseSpace,TDenseSp
-     * @param b: RHS Vector
+     * @brief Performing the update of the solution
+     * @details Incremental update within newton iteration. It updates the state variables at the end of the time step u_{n+1}^{k+1}= u_{n+1}^{k}+ \Delta u
+     * @param rModelPart The model of the problem to solve
+     * @param rDofSet Set of all primary variables
+     * @param A LHS matrix
+     * @param Dx incremental update of primary variables
+     * @param b RHS Vector
      */
 
     void Update(
         ModelPart& rModelPart,
         DofsArrayType& rDofSet,
         TSystemMatrixType& A,
         TSystemVectorType& Dx,
-        TSystemVectorType& b ) override
+        TSystemVectorType& b 
+        ) override
     {
         KRATOS_TRY;
 
-        // std::cout << " Update " << std::endl;
-
-        const unsigned int NumThreads = OpenMPUtils::GetNumThreads();
-
         // Update of displacement (by DOF)
-        OpenMPUtils::PartitionVector DofPartition;
-        OpenMPUtils::DivideInPartitions(rDofSet.size(), NumThreads, DofPartition);
-
-        const int ndof = static_cast<int>(rDofSet.size());
-        typename DofsArrayType::iterator DofBegin = rDofSet.begin();
+        const int num_dof = static_cast<int>(rDofSet.size());
 
-        #pragma omp parallel for firstprivate(DofBegin)
-        for(int i = 0;  i < ndof; i++)
-        {
-            typename DofsArrayType::iterator itDof = DofBegin + i;
+        #pragma omp parallel for
+        for(int i = 0;  i < num_dof; ++i) {
+            auto it_dof = rDofSet.begin() + i;
 
-            if (itDof->IsFree() )
-            {
-                itDof->GetSolutionStepValue() += TSparseSpace::GetValue(Dx,itDof->EquationId());
-            }
+            if (it_dof->IsFree())
+                it_dof->GetSolutionStepValue() += TSparseSpace::GetValue(Dx,it_dof->EquationId());
         }
-
+        
         // Updating time derivatives (nodally for efficiency)
-        OpenMPUtils::PartitionVector NodePartition;
-        OpenMPUtils::DivideInPartitions(rModelPart.Nodes().size(), NumThreads, NodePartition);
-
-        const int nnodes = static_cast<int>(rModelPart.Nodes().size());
-        NodesArrayType::iterator NodeBegin = rModelPart.Nodes().begin();
-
-        #pragma omp parallel for firstprivate(NodeBegin)
-        for(int i = 0;  i < nnodes; i++)
-        {
-            array_1d<double, 3 > DeltaDisplacement;
-
-            NodesArrayType::iterator itNode = NodeBegin + i;
+        const int num_nodes = static_cast<int>(rModelPart.Nodes().size());
 
-            noalias(DeltaDisplacement) = (itNode)->FastGetSolutionStepValue(DISPLACEMENT) - (itNode)->FastGetSolutionStepValue(DISPLACEMENT, 1);
-
-            array_1d<double, 3 > & CurrentVelocity            = (itNode)->FastGetSolutionStepValue(VELOCITY, 0);
-            const array_1d<double, 3 > & PreviousVelocity     = (itNode)->FastGetSolutionStepValue(VELOCITY, 1);
+        #pragma omp parallel for
```

Comment:
this is a good change. however WHY IS THIS PART OF THIS PR?

it is a file used by a lot of people so please move this to its own PR

Reply:
The problem is that I create the implicit_scheme in order to reduce code duplication. I added two tests, one in python and other in cpp, everything is right